rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/impl/RepositoryGraphImpl.java

```diff
@@ -0,0 +1,164 @@
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j.impl;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.commons.rdf.api.BlankNodeOrIRI;
+import org.apache.commons.rdf.api.Graph;
+import org.apache.commons.rdf.api.IRI;
+import org.apache.commons.rdf.api.RDFTerm;
+import org.apache.commons.rdf.api.Triple;
+import org.apache.commons.rdf.rdf4j.RDF4JGraph;
+import org.apache.commons.rdf.rdf4j.RDF4JTriple;
+import org.eclipse.rdf4j.common.iteration.Iterations;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryResult;
+
+public class RepositoryGraphImpl extends AbstractRepositoryGraphLike<Triple> implements Graph, RDF4JGraph {
+
+	private final Resource[] contextFilter;
+
+	public RepositoryGraphImpl(Repository repository, boolean includeInferred, boolean unionGraph) {
+		super(repository, includeInferred);
+		if (unionGraph) {
+			// no context filter aka any context
+			this.contextFilter = new Resource[] { };
+		} else {
+			// default context: null
+			this.contextFilter = new Resource[] { null };
+		}
+	}
+
+	public RepositoryGraphImpl(Repository repository, boolean includeInferred, Resource... contextFilter) {
+		super(repository, includeInferred);
+		this.contextFilter = contextFilter;
+	}
+
+
+	@Override
+	public void add(Triple tripleLike) {
+		Statement statement = rdf4jTermFactory.asStatement(tripleLike);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.add(statement, contextFilter);
+			conn.commit();
+		}
+	}
+
+
+	@Override
+	public boolean contains(Triple tripleLike) {
+		Statement statement = rdf4jTermFactory.asStatement(tripleLike);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			return conn.hasStatement(statement, includeInferred, contextFilter);
+		}
+	}
+
+	@Override
+	public void remove(Triple tripleLike) {
+		Statement statement = rdf4jTermFactory.asStatement(tripleLike);
+		try (RepositoryConnection conn = getRepositoryConnection()) {
+			conn.remove(statement, contextFilter);
+			conn.commit();
```

Comment:
Perhaps some kind of `ThreadLocal` or transaction argument..? 

Or ask for a `TransactionGraph` that only does `.commit()` on `.close()`? 

Is a `RepositoryConnection` thread safe, or would I need `ThreadLocal`s?

I think it might be weird if the `Graph` out of the box didn't autocommit as the other implementations do - so optimizing for faster should be something you do explicitly (Commons RDF does not know how you manage your threads)